src/sqlfluff/dialects/dialect_mysql.py

```diff
@@ -577,6 +568,19 @@ class IntervalExpressionSegment(BaseSegment):
     ),
 )
 
+"""Format for user and role names. Overwrite SingleIdentifierGrammar below to allow
+two single_identifier_segment separated by an at sign.
+    https://dev.mysql.com/doc/refman/8.0/en/account-names.html
+    https://dev.mysql.com/doc/refman/8.0/en/role-names.html
+"""
+single_identifier_segment = ansi_dialect.get_grammar("SingleIdentifierGrammar").copy(
+    insert=[
+        Ref("SessionVariableNameSegment"),
+        Ref("SingleQuotedIdentifierSegment"),
+        Ref("DoubleQuotedLiteralSegment"),
+    ]
+)
```

Comment:
I feel I should have ended up with a `SingleIdentifierSegment` that I can `Ref` below, but I'm not sure how to do this with `get_grammar`.

Reply:
This works:

```python
class SingleIdentifierSegment(BaseSegment):
    """A SingleIdentifierSegment. Override the default SingleIdentifierGrammar to allow
    two of these segments separated by an at sign.

    https://dev.mysql.com/doc/refman/8.0/en/account-names.html
    https://dev.mysql.com/doc/refman/8.0/en/role-names.html
    """

    type = "single_identifier"
    match_grammar: Matchable = ansi_dialect.get_grammar("SingleIdentifierGrammar").copy(
        insert=[
            Ref("SessionVariableNameSegment"),
            Ref("SingleQuotedIdentifierSegment"),
            Ref("DoubleQuotedLiteralSegment"),
        ]
    )
```

but it diffs all the MySQL YML test files.

Reply:
That's fine. The point of the YAMLs being in git, is so we can ensure there are no unexpected changes. This is expected.